functions/src/adapter.ts

```diff
@@ -49,8 +49,11 @@ export function adaptDocument(
 ): Record<string, any> {
   const fields = getFieldsToIndex()
   const data = snapshot.data()
+  if (data && '_firestore_id' in data) {
+    delete data.id
+  }
   if (fields.length === 0 || !data) {
```

Comment:
What are you testing with:
```
  !data
```

Because it does not work with `[]` for example as `[]` returns true.

Reply:
I have added `!data` to avoid an error in the following condition: ` adaptValues(key, data[key])`

Reply:
is `{}` correct?